### ISSUE TITLE
How to organize human feedback results into a JSON file?

### ISSUE THREAD

**Issue by Mr-Loevan** [yk7333/d3po#14]
Thanks for your great work!
I am confused about the `.json` file (i.e. human feedback results), 
it's an array like `[0, -1, 0, 0, -1, 0, -1]`
Does 0 denote preferred image and -1 denote dis-perferred image?

**Comment by yk7333** [comment]
Yes, you need to set a higher value for the better images than the worse ones. For example, both `[1,0,1,1,0,1,0]` and `[0, -1, 0, 0, -1, 0, -1]` are acceptable and equivalent. For tasks that are not simply binary (0 or 1), such as rating the attractiveness of an image, you can also assign multiple levels of scores, such as `[2,0,-1,3,-2,0]`. Samples with different scores can all be learned, leading to better training results.

**Comment by Mr-Loevan** [comment]
> Yes, you need to set a higher value for the better images than the worse ones. For example, both `[1,0,1,1,0,1,0]` and `[0, -1, 0, 0, -1, 0, -1]` are acceptable and equivalent. For tasks that are not simply binary (0 or 1), such as rating the attractiveness of an image, you can also assign multiple levels of scores, such as `[2,0,-1,3,-2,0]`. Samples with different scores can all be learned, leading to better training results.

Thanks for your reply! And could it be possible to directly apply D3PO in SD_XL ?

**Comment by yk7333** [comment]
Certainly, the D3PO method can be applied to any diffusion structure. Wish you success in your research!
